database/node_registry.py

```diff
@@ -0,0 +1,48 @@
+import os
+import requests
+import json
+
+
+class NodeRegistry:
+    def read_json(self, path):
+        # project_root = "node_manager"
+        # settings_path = os.path.join(project_root, 'settings.json')
+        with open(path, 'r') as f:
+            json_content = json.load(f)
+
+        return json_content
+
+    def write_json(self, path, data):
+        f = open(path, 'w')
```

Comment:
Always use with statements when opening files. This can leave a file open indefinitely

Reply:
With statement added